File: frontend/src/lib/api/leaderboard.ts
import { api } from "@/lib/api/axios"
import { LeaderboardResponse } from "@/types/api"

export const leaderboardApi = {
  getLeaderboard: async (): Promise<LeaderboardResponse> => {
    const { data } = await api.get<LeaderboardResponse>("/leaderboard")
    return data
  },
}

File: app/api/routes/leaderboard.py
# app/api/routes/leaderboard.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.schemas.leaderboard import LeaderboardResponse
from app.services import leaderboard_service

router = APIRouter(prefix="/leaderboard", tags=["leaderboard"])


@router.get("", response_model=LeaderboardResponse)
def get_leaderboard(db: Session = Depends(get_db)) -> LeaderboardResponse:
    return leaderboard_service.get_leaderboard(db)